vote.py

```diff
@@ -0,0 +1,190 @@
+from brutal.core.plugin import BotPlugin, cmd
+from brutal.core.utils import split_args_by
+
+
+class Vote(BotPlugin):
+    VOTING_NONE = 0
+    VOTING_YESNO = 1
+    VOTING_QUESTION = 2
+
+    def setup(self, *args, **kwargs):
+        self.reset()
+
+    def reset(self):
+        """Reset all voting variables"""
+        self.voting = self.VOTING_NONE
+        self.yeses = 0
+        self.nos = 0
+        self.voted = {}
+
+    def voting_in_progress(self, question=False):
+        if question:
+            return self.voting == self.VOTING_QUESTION
+        return self.voting is not self.VOTING_NONE
+
+    def judgement(self, event=None, penalty=''):
+        """Summarize votes and make decision"""
+        # TODO: if voting succeeded, get op and punish user
+        msg = 'unsuccessfully'
+        if self.yeses + self.nos > 1 and self.yeses > self.nos:
+            msg = 'successfully '
+
+        msg += ' with {0}x yes, {1}x no'.format(self.yeses, self.nos)
+        result = '{0} {1}'.format(penalty, event.args[0])
+
+        self.reset()
+
+        return 'Voting to {0} finished {1}'.format(result, msg)
+
+    def vote_result(self, event=None, penalty=''):
+        """Start voting of given penalty"""
+        if self.voting_in_progress() or self.voting_in_progress(question=True):
+            return 'Voting in progress...'
+
+        args = event.args
+        if not len(args):
+            return 'No user specified'
+
+        if '-' in args:
+            args = split_args_by(args, '-')
+
+        try:
+            time = int(args[1])
+        except:
+            time = 60
+
+        user = args[0]
+        self.voting = self.VOTING_YESNO
+
+        self.delay_task(time, self.judgement, event=event, penalty=penalty)
+
+        return 'Voting to {0} {1} started for {2} secs. (!yes / !no)' \
+            .format(penalty, user, time)
+
+    # TODO: put all these commands into one regexp match
+    @cmd
+    def votekick(self, event):
+        return self.vote_result(event, 'kick')
+
+    @cmd
+    def voteban(self, event):
+        return self.vote_result(event, 'ban')
+
+    @cmd
+    def voteunban(self, event):
+        return self.vote_result(event, 'unban')
+
+    @cmd
+    def votemute(self, event):
+        return self.vote_result(event, 'mute')
+
+    def yesno_result(self, event=None):
+        """Return results of yes/no question"""
+        yes = []
+        no = []
+        for k, v in self.voted.iteritems():
+            if v[1]:
```

Comment:
There're class variables called self.yeses and self.nos, which take care of counting yeses/nos. What I'm doing here is checking whether user voted yes/no